com.unity.shadergraph/Editor/ShaderFoundry/Linker/Properties/PropertyDeclarations.cs

```diff
@@ -0,0 +1,382 @@
+using System;
+using System.Collections.Generic;
+
+namespace UnityEditor.ShaderFoundry
+{
+    class PropertyDeclarations
+    {
+        static Dictionary<string, Func<FieldPropertyContext, FieldPropertyData, bool>> TypeHandlers = new Dictionary<string, Func<FieldPropertyContext, FieldPropertyData, bool>>
+        {
+            {"bool", HandleBoolType },
+            {"int", HandleIntegerType },
+            {"float", HandleFloatType },
+            {"float2", HandleFloat2Type },
+            {"float3", HandleFloat3Type },
+            {"float4", HandleFloat4Type },
+            {"UnityTexture2D", HandleUnityTexture2D },
+            {"UnityTexture2DArray", HandleUnityTexture2DArray },
+            {"UnityTextureCube", HandleUnityTextureCube },
+            {"UnityTexture3D", HandleUnityTexture3D },
+            {"VTPropertyWithTextureType", HandleVirtualTexture },
+        };
+
+        static Dictionary<string, Action<ShaderAttribute, FieldPropertyContext>> AttributeHandlers = new Dictionary<string, Action<ShaderAttribute, FieldPropertyContext>>
+        {
+            {"Gamma", DefaultAttributeProcessor},
+            {"HDR", DefaultAttributeProcessor},
+            {"HideInInspector", DefaultAttributeProcessor},
+            {"MainTexture", DefaultAttributeProcessor},
+            {"MainColor", DefaultAttributeProcessor},
+            {"NoScaleOffset", DefaultAttributeProcessor},
+            {"Normal", DefaultAttributeProcessor},
+            {"PerRendererData", DefaultAttributeProcessor},
+            {"KeywordEnum", DefaultAttributeProcessor},
+            {"Toggle", DefaultAttributeProcessor},
+            {"ToggleOff", DefaultAttributeProcessor},
+            {"Enum", DefaultAttributeProcessor},
+            {"PowerSlider", DefaultAttributeProcessor},
+            {"IntRange", DefaultAttributeProcessor},
+            {"Space", DefaultAttributeProcessor},
+            {"Header", DefaultAttributeProcessor},
+            {"VTexLayers", DefaultAttributeProcessor },
+        };
+
```

Comment:
wondering if it makes sense to have a set of static strings for the keywords used in the methods below (not necessarily a data container of strings, but just a common definition that gets assigned for use rather than having all of the string literals duplicated in the different methods).